packages/beacon-node/src/chain/opPools/aggregatedAttestationPool.ts

```diff
@@ -426,15 +427,15 @@ export function isValidAttestationData(
 
   if (!ssz.phase0.Checkpoint.equals(data.source, justifiedCheckpoint)) return false;
 
-  // Shuffling can't have changed if we're in the first few epochs
-  if (stateEpoch < 2) {
-    return true;
-  }
   // Otherwise the shuffling is determined by the block at the end of the target epoch
   // minus the shuffling lookahead (usually 2). We call this the "pivot".
-  const pivotSlot = computeStartSlotAtEpoch(targetEpoch - 1) - 1;
-  const statePivotBlockRoot = toHexString(getBlockRootAtSlot(state, pivotSlot));
+  const pivotSlot = computeEndSlotAtEpoch(targetEpoch - MIN_SEED_LOOKAHEAD);
 
+  if (pivotSlot < 0) {
```

Comment:
it's too relaxed to have this condition as a `stateEpoch` 1000 can include attestations from epoch 1

as the issue only happens for stateEpoch 2 and attestations in epoch 1 (I suppose?) I think we can just change the condition above to:

```typescript
if (stateEpoch <= 2) {
  return true;
}
```

Reply:
I think we would rather check that difference between `targetEpoch` and `stateEpoch` is not greater than `MIN_SEED_LOOKAHEAD + 1`. This makes more descriptive approach than hardcoding the `2` epoch. 

```ts

if (targetEpoch < stateEpoch - (MIN_SEED_LOOKAHEAD + 1)) {
   return false; 
}
```

Reply:
Lighthouse check this condition as follows. 

```rs
let pivot_slot =
            if target_epoch == state.previous_epoch() || target_epoch == state.current_epoch() {
                (target_epoch - shuffling_lookahead).end_slot(slots_per_epoch)
            } else {
                return false;
            };
```

https://github.com/sigp/lighthouse/blob/aa022f46855df2a1420a6a80a788c73dc2779aa7/beacon_node/beacon_chain/src/beacon_chain.rs#L2002

Reply:
@dapplion What's your opinion on this?

Reply:
> it's too relaxed to have this condition as a stateEpoch 1000 can include attestations from epoch 1

I missed the 1st check to make sure `targetEpoch` is only current epoch or previous epoch

so I think all ways below work

```typescript
  if (stateEpoch <= 2) {
    return true;
  }
```

or 

```typescript
  if (stateEpoch <= MIN_SEED_LOOKAHEAD + 1) {
    return true;
  }
```
or

```typescript
  if (pivotSlot < 0) {
    return true;
  }
```

I'm fine with whatever way, as long as we have a good comment (I prefer 1 & 2 thru)